src/inspector_socket_server.cc

```diff
@@ -376,30 +383,44 @@ void InspectorSocketServer::SendListResponse(InspectorSocket* socket) {
 
 bool InspectorSocketServer::Start() {
   CHECK_EQ(state_, ServerState::kNew);
-  sockaddr_in addr;
-  uv_tcp_init(loop_, &server_);
-  uv_ip4_addr(host_.c_str(), port_, &addr);
-  int err = uv_tcp_bind(&server_,
-                        reinterpret_cast<const struct sockaddr*>(&addr), 0);
-  if (err == 0)
-    err = GetPort(&server_, &port_);
-  if (err == 0) {
-    err = uv_listen(reinterpret_cast<uv_stream_t*>(&server_), 1,
-                    SocketConnectedCallback);
+  struct addrinfo hints;
+  memset(&hints, 0, sizeof(struct addrinfo));
+  hints.ai_flags = AI_NUMERICSERV;
+  hints.ai_socktype = SOCK_STREAM;
+  uv_getaddrinfo_t req;
+  const std::string port_string = std::to_string(port_);
+  int err = uv_getaddrinfo(loop_, &req, nullptr, host_.c_str(),
+                           port_string.c_str(), &hints);
+  if (err < 0) {
+    if (out_ != NULL) {
+      fprintf(out_, "Unable to resolve \"%s\": %s\n", host_.c_str(),
+              uv_strerror(err));
+    }
+    return false;
   }
-  if (err == 0 && connected_sessions_.empty()) {
-    state_ = ServerState::kRunning;
-    PrintDebuggerReadyMessage(host_, port_, delegate_->GetTargetIds(), out_);
+  for (addrinfo* address = req.addrinfo; address != nullptr;
+       address = address->ai_next) {
+    err = ServerSocket::Listen(this, address->ai_addr, loop_);
```

Comment:
We are not printing out resolved address. Instead localhost:port will be printed. Only one port will be outputted - which is only a problem when port set to 0.